src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,97 @@
-          </button>
-        </td>
-      </tr>
+export const TodoList: React.FC<T> = ({
+  input,
+  todos,
+  filterType,
+  isDedicatedUser,
+  setIsDedicatedUser,
+  show,
+  setShow,
+}) => {
+  const filterTodos = (array: Todo[], type: string) => {
+    switch (type) {
+      case 'active':
+        return array.filter(todo => !todo.completed);
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">1</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+      case 'completed':
+        return array.filter(todo => todo.completed);
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">6</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">
-            qui ullam ratione quibusdam voluptatem quia omnis
-          </p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+      default:
+        return array;
+    }
+  };
+
+  const onClick = (id: number) => {
+    setIsDedicatedUser(id);
+    setShow(true);
+  };
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">8</td>
-        <td className="is-vcentered">
-          <span className="icon" data-cy="iconCompleted">
-            <i className="fas fa-check" />
-          </span>
-        </td>
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-success">quo adipisci enim quam ut ab</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
+  const filter = filterTodos(todos, filterType);
+
+  // ігнорувати регістр при фільтрації
+
+  return (
+    <table className="table is-narrow is-fullwidth">
+      <thead>
+        <tr>
+          <th>#</th>
+          <th>
             <span className="icon">
-              <i className="far fa-eye" />
+              <i className="fas fa-check" />
             </span>
-          </button>
-        </td>
-      </tr>
-    </tbody>
-  </table>
-);
+          </th>
+          <th>Title</th>
+          <th> </th>
+        </tr>
+      </thead>
+
+      <tbody>
+        {filter.map(todo => todo.title.toLowerCase()
+          .includes(input.toLowerCase())
+          && (
+            <tr data-cy="todo" key={todo.id} className="">
```

Comment:
Use a destruction for `todo`

Reply:
{input !== '' && (
I left it because when changes are made, it enters 0 near the input. I also can’t transfer filtering to a variable since the layout depends on it